bootstrap/v2/pkg/apis/apps/kfdef/v1alpha1/application_types.go

```diff
@@ -134,6 +214,12 @@ type AppConfig struct {
 // KfDefStatus defines the observed state of KfDef
 type KfDefStatus struct {
 	Conditions []KfDefCondition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,6,rep,name=conditions"`
+	// ReposCache is used to cache information about local caching of the URIs.
+	ReposCache map[string]RepoCache `json:"reposCache,omitempty"`
```

Comment:
Status is where we track temporary information.

Suppose we check out the app on different machines. On both machines we should be able to cache any repos by fetching the source 
https://github.com/kubeflow/kubeflow/archive/master.zip

And storing it in a directory like .cache.

But the .cache directory will be different on different machines. So following K8s conventions it should be stored in status because its temporary/ephmeral information as opposed to describing the desired state of the world.

Reply:
So we will re-fetch the repo sometime? when will this happen?

Reply:
When code requires a local copy of the repos it would ideally call SyncCache. 
#2959 is the issue to download the cache if it is missing.